src/V3File.cpp

```diff
@@ -976,6 +982,7 @@ class VIdProtectImp final {
     ~VIdProtectImp() = default;
     // METHODS
     string passthru(const string& old) {
+        const std::lock_guard<std::mutex> lock{m_mutex};
```

Comment:
Claim the mutex after the check for the protectIds option below that short circuit returns (which is the common case), please check elsewhere for similar.